### PR TITLE
Merging outgoing donation

### LINKED ISSUE #18: createOutgoingDonation Endpoint + DB Set up
- the createOutgoingDonation API endpoint 
- create OutgoingDonation Table 
- create OrderDetails Table

### PR DESCRIPTION
**Pull request by Kaleab-A**

### CODE REVIEW

File: MTM-Backend/prisma/migrations/20231111192652_updated_tables/migration.sql
Comment (OliviaLiu417) [inline MTM-Backend/prisma/migrations/20231111192652_updated_tables/migration.sql:1]: Let's add the migration folder to gitignore and not push to the repo

File: MTM-Backend/prisma/schema.prisma
Comment (OliviaLiu417) [inline MTM-Backend/prisma/schema.prisma:1]: Code format using prettier

File: MTM-Backend/src/types/donation.ts
Comment (OliviaLiu417) [inline MTM-Backend/src/types/donation.ts:1]: Maybe make it clearer what differences are between DonationDetailsType and DonationDetailType and what purposes they serve?

File: MTM-Backend/src/routes/v1/donation/donation.service.ts
Comment (OliviaLiu417) [inline MTM-Backend/src/routes/v1/donation/donation.service.ts:1]: Code format

File: MTM-Backend/src/types/donation.ts
Comment (Kaleab-A) [reply MTM-Backend/src/types/donation.ts:1]: When looping through DonationDetails, each element is DonatoinDetail. So I created a type for both.

Comment (OliviaLiu417) [reply MTM-Backend/src/types/donation.ts:1]: Maybe comment it in the code

Comment (Kaleab-A) [reply MTM-Backend/src/types/donation.ts:1]: okay

File: MTM-Backend/src/routes/v1/donation/donation.router.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+const donationRouter = express.Router();
+
+const createOutgoingDonation = async (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) => {
+  try {
+    console.log(req.body.userId);
```
Comment (OliviaLiu417) [inline MTM-Backend/src/routes/v1/donation/donation.router.ts]: remove the console logs

File: MTM-Backend/src/routes/v1/donation/donation.router.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      async (donationDetail: DonationDetailType) => {
+        const newDonationDetail = await DonationService.createDonationDetails(
+          donationDetail.itemId,
+          newDonation.id,
+          donationDetail,
+        );
+      },
+    );
+
+    console.log(
```
Comment (OliviaLiu417) [inline MTM-Backend/src/routes/v1/donation/donation.router.ts]: remove console log

File: MTM-Backend/prisma/schema.prisma [hunk 3]
```diff
@@ -2,20 +2,70 @@
...
 generator client {
-  provider = "prisma-client-js"
+    provider = "prisma-client-js"
 }
 
 datasource db {
-  provider = "mysql"
-  url      = env("DATABASE_URL")
+    provider = "mysql"
+    url      = env("DATABASE_URL")
```
Comment (OliviaLiu417) [inline MTM-Backend/prisma/schema.prisma]: code formats

Comment (Kaleab-A) [reply MTM-Backend/prisma/schema.prisma]: Prettier isn't changing the current formatting for schema.prisma. I sent an image on slack.

**Review by OliviaLiu417 — APPROVED** [review]

**Pull request merged**